plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTable.java

```diff
@@ -180,7 +187,7 @@ public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, Connect
                 .useSnapshot(snapshotId.get())
                 .includeColumnStats();
         // TODO make the cursor lazy
-        return buildRecordCursor(getStatisticsByPartition(tableScan), icebergTable.spec().fields());
+        return buildRecordCursor(getStatisticsByPartition(tableScan), getAllPartitionFields(icebergTable));
```

Comment:
Isn't there a class level variable with these already?
```suggestion
        return buildRecordCursor(getStatisticsByPartition(tableScan), partitionFields);
```